### PR TITLE
feat: implemented onBeforeTextEdit and onBeforeTextSubmit callbacks

### LINKED ISSUE #3779: onBeforeTextEdit and onBeforeTextSubmit callbacks
Hi! Can I capture the event when the user starts to edit a text element?
I want change the text right at the point in time when the edit starts, and change it again when the edit finishes.

My use-case is to implement native markdown links that render nicely... i.e. you click a text element and Obsidian opens another drawing or it opens a document. This link feature already works. 
What I want now is that while editing, the text reads "[[This is a link|alias]]" i.e. with full markdown markup, but when the user finishes editing the text is parsed and changed to "[[alias]]" or just "alias" or ":link:alias".

I can implement my own text element editor overlay, or I can implement ALT+click to toggle a text element between markup and rendered view, thus the user first ALT+clicks, then double clicks to edit the markup, then ALT+clicks again to toggle the text element to rendered view... but the nicest would be if I could capture the edit event and alter the text at that point in time.

Any suggestion or idea is most welcome!

_Originally posted by @zsviczian in https://github.com/excalidraw/excalidraw/discussions/3776_

### PR DESCRIPTION
**Pull request by zsviczian**
Would these changes do the trick to resolve #3779?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "@types/react-dom": "17.0.3",
     "@types/socket.io-client": "1.4.36",
     "browser-fs-access": "0.18.0",
+    "canvas": "2.8.0",
```
Comment (ad1992) [inline package.json:31]: looks like this was pushed by mistake, can you remove it ?

Comment (zsviczian) [reply package.json]: done

**Comment by zsviczian** [comment]
Hi @ad1992: Do I need to do anything further with this PR? What does it depend on whether this PR will be merged into the next release or not? How can I help the process?

**Comment by ad1992** [comment]
> Hi @ad1992: Do I need to do anything further with this PR? What does it depend on whether this PR will be merged into the next release or not? How can I help the process?

Hey @zsviczian  you don't need to make any more changes in this PR as of now, can you revert `yarn.lock`? There might be some API changes that are not finalized yet so will merge once the changes are finalized. Should be able to merge it over the weekend.

**Comment by zsviczian** [comment]
> > Hi @ad1992: Do I need to do anything further with this PR? What does it depend on whether this PR will be merged into the next release or not? How can I help the process?
> 
> Hey @zsviczian you don't need to make any more changes in this PR as of now, can you revert `yarn.lock`? There might be some API changes that are not finalized yet so will merge once the changes are finalized. Should be able to merge it over the weekend.

Sorry... I am not familiar enough with git. I now deleted yarn.lock while looking for the action to revert 😳. How?

**Pull request closed without merging**